cmd/soroban-rpc/internal/test/metrics_test.go

```diff
@@ -25,6 +28,23 @@ func TestMetrics(t *testing.T) {
 		config.Version,
 	)
 	require.Contains(t, metrics, buildMetric)
+
+	logger := test.daemon.Logger()
+	err := errors.Errorf("test-error")
+	logger.WithError(err).Error("test error 1")
+	logger.WithError(err).Error("test error 2")
+
+	metricFamilies, err := test.daemon.MetricsRegistry().Gather()
+	assert.NoError(t, err)
+	var metric *io_prometheus_client.MetricFamily
+	for _, mf := range metricFamilies {
+		if *mf.Name == "soroban_rpc_log_error_total" {
+			metric = mf
+			break
+		}
+	}
+	val := metric.Metric[0].Counter.GetValue()
+	assert.Equal(t, val, 2.0)
```

Comment:
There might be some other parts of the code which log errors when the daemon is initialized in `NewTest(t, nil)`. and in that case the `soroban_rpc_log_error_total ` metric could be greater than 2. So, it would be more robust if this test asserted that value was positive rather than exactly 2

Reply:
Ah yeah nice catch. I can add a GreaterOrEqual assertion here.